disperser/common/chunkstore/chunk_writer.go

```diff
@@ -0,0 +1,103 @@
+package chunkstore
+
+import (
+	"context"
+	"fmt"
+	"github.com/Layr-Labs/eigenda/common/aws/s3"
+	"github.com/Layr-Labs/eigenda/disperser"
+	"github.com/Layr-Labs/eigenda/encoding"
+	"github.com/Layr-Labs/eigenda/encoding/rs"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	"github.com/consensys/gnark-crypto/ecc/bn254"
+)
+
+// ChunkCoefficientMetadata contains metadata about how chunk coefficients are stored.
+// Required for reading chunk coefficients using ChunkReader.GetChunkCoefficients().
+type ChunkCoefficientMetadata struct {
+	// The total size of file containing all chunk coefficients for the blob.
+	DataSize int
+	// The maximum fragment size used to store the chunk coefficients.
+	FragmentSize int
```

Comment:
Could we be more specific here and say its uint64 or what's appropriate than generic int type ?

Reply:
Will do. Let's go with `uint32`.

As a side note, one of the quirks of golang that drives me up a wall is how they strongly encourage everybody to use the `int` type everywhere. For example, why does `len(x)` return a signed value? If I were in charge of the language design, I'd never have supported the types `int` and `uint` in the first place. /rant